Source/WebKit/UIProcess/WebProcessPool.cpp

```diff
@@ -2681,4 +2686,58 @@ void WebProcessPool::updateWebProcessSuspensionDelayWithPacing(WeakHashSet<WebPr
 
 #endif // ENABLE(WEB_PROCESS_SUSPENSION_DELAY)
 
+#if ENABLE(CONTENT_EXTENSIONS)
+
+constexpr static Seconds resourceMonitorRuleListCheckInterval = 24_h;
+
+WebCompiledContentRuleList* WebProcessPool::cachedResourceMonitorRuleList()
+{
+    if (!m_resourceMonitorRuleListCache)
+        loadOrUpdateResourceMonitorRuleList();
+
+    return m_resourceMonitorRuleListCache.get();
+}
+
+void WebProcessPool::loadOrUpdateResourceMonitorRuleList()
+{
+    if (m_resourceMonitorRuleListLoading)
+        return;
+
+    WEBPROCESSPOOL_RELEASE_LOG(Process, "loadOrUpdateResourceMonitorRuleList: rule list is requested");
+
+    m_resourceMonitorRuleListLoading = true;
+
+    platformLoadResourceMonitorRuleList([weakThis = WeakPtr { *this }] {
+        RefPtr protectedThis = weakThis.get();
+        if (!protectedThis)
+            return;
+
+        protectedThis->m_resourceMonitorRuleListLoading = false;
+
+        RefPtr ruleList = protectedThis->m_resourceMonitorRuleListCache;
+        if (!ruleList)
+            return;
```

Comment:
`platformLoadResourceMonitorRuleList` returns nullptr all the time on non-Darwin platforms. That means that every single time `cachedResourceMonitorRuleList()` is called, it will do some unnecessary work to load an empty rule list.

It seems like it would be nice to add some logic to only call `platformLoadResourceMonitorRuleList` at most once per day.